Ryujinx/Updater/Updater.cs

```diff
@@ -0,0 +1,292 @@
+using Gtk;
+using ICSharpCode.SharpZipLib.GZip;
+using ICSharpCode.SharpZipLib.Tar;
+using ICSharpCode.SharpZipLib.Zip;
+using Newtonsoft.Json.Linq;
+using Ryujinx.Common.Logging;
+using Ryujinx.Ui;
+using System;
+using System.IO;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Ryujinx
+{
+    public static class Updater
+    {
+        internal static bool Running;
+
+        private static readonly string HomeDir          = AppDomain.CurrentDomain.BaseDirectory;
+        private static readonly string UpdateDir        = Path.Combine(Path.GetTempPath(), "Ryujinx", "update");
+        private static readonly string UpdatePublishDir = Path.Combine(Path.GetTempPath(), "Ryujinx", "update", "publish");
+
+        private static string _jobId;
+        private static string _buildVer;
+        private static string _platformExt;
+        private static string _buildUrl;
+
+        private const string MasterUrl = "https://ci.appveyor.com/api/projects/gdkchan/ryujinx/branch/master";
+
+        public async static void BeginParse(MainWindow mainWindow, bool showVersionUpToDate)
+        {
+            if (Running) return;
+            Running = true;
+            mainWindow.UpdateMenuItem.Sensitive = false;
+
+            // Detect current platform
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                _platformExt = "osx_x64.zip";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                _platformExt = "win_x64.zip";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                _platformExt = "linux_x64.tar.gz";
+            }
+
+            // Get latest version number from Appveyor
+            try
+            {
+                using (WebClient jsonClient = new WebClient())
+                {
+                    string fetchedJson = await jsonClient.DownloadStringTaskAsync(MasterUrl);
```

Comment:
```suggestion
                    string  fetchedJson = await jsonClient.DownloadStringTaskAsync(MasterUrl);
```
and align other lines